spec/factories/vm_or_template.rb

```diff
@@ -14,10 +14,59 @@
     raw_power_state "never"
   end
 
-  factory(:vm,             :class => "Vm",            :parent => :vm_or_template)
-  factory(:vm_cloud,       :class => "VmCloud",       :parent => :vm)       { cloud true }
-  factory(:vm_infra,       :class => "VmInfra",       :parent => :vm)
+  factory(:vm, :class => "Vm", :parent => :vm_or_template)
+
+  factory(:vm_orphaned, :class => "Vm", :parent => :vm_or_template) do
+    ext_management_system nil
+    storage_id            123
```

Comment:
@fryguy would you rather use traits for the `with_storage`?
That way, `:with_storage` could be specified all throughout the tree and we wouldn't need a `*_with_storage` version of all of these